pkg/asset/machines/gcp/gcpmachines_test.go

```diff
@@ -149,6 +159,26 @@ func getICWithSecureBoot() *installconfig.InstallConfig {
 	return ic
 }
 
+func getICWithServiceAccount() *installconfig.InstallConfig {
+	ic := getBaseInstallConfig()
+	ic.Config.Platform.GCP.DefaultMachinePlatform = &gcptypes.MachinePool{ServiceAccount: "user-service-account@some-project.iam.gserviceaccount.com"}
+	return ic
+}
+
+func getICWithServiceAccountControlPlaneMachine() *installconfig.InstallConfig {
+	ic := getBaseInstallConfig()
+	ic.Config.Platform.GCP.DefaultMachinePlatform = &gcptypes.MachinePool{ServiceAccount: "user-service-account@some-project.iam.gserviceaccount.com"}
+	ic.Config.ControlPlane = &types.MachinePool{
+		Name:     "master",
```

Comment:
> nit: Should we change this to control-plane?

Since that string is used in getBaseGCPMachine() that would require a change to all of the tests.  As its just a string I don't think its necessary, or if want to change that it should be in a separate PR.